velox/type/Timestamp.cpp

```diff
@@ -225,15 +229,33 @@ bool Timestamp::epochToUtc(int64_t epoch, std::tm& tm) {
   }
   tm.tm_year = y;
   tm.tm_yday = days;
-  auto* ip = monthLengths[leapYear];
-  for (tm.tm_mon = 0; days >= ip[tm.tm_mon]; ++tm.tm_mon) {
-    days = days - ip[tm.tm_mon];
-  }
-  tm.tm_mday = days + 1;
+  auto* months = daysBeforeFirstDayOfMonth[leapYear];
+  tm.tm_mon = std::upper_bound(months, months + 12, days) - months - 1;
+  tm.tm_mday = days - months[tm.tm_mon] + 1;
   tm.tm_isdst = 0;
   return true;
 }
 
+bool Timestamp::utcToEpoch(const std::tm& tm, int64_t& seconds) {
```

Comment:
the functions always return true? Why not making it return the seconds/epoch?

Reply:
When `decltype(tm.tm_year)` is `int32_t` (`static_assert`ed below), it cannot fail.  I leave out the return value just in case if we want to port to some platform where `decltype(tm.tm_year)` is `int64_t` in the future.

Reply:
But even in that case, shouldn't it just throw? Or even just return `Expected<int64_t>`. Output parameters always make the API more awkward to use and are less obvious.

Reply:
We usually don't want to throw in low level function like this.  I will change it to return seconds for now and we can refactor the signature later if we ever want to port.